### PR TITLE
ambient: fix: use lastest pod status to determine the action

### PR DESCRIPTION
**Pull request by PlatformLC**
**Please provide a description of this PR:**
numerous err/warning log outputs unexpected while label/unlabel the ambient mode :
```
2024-02-03T13:43:47.988707Z    warn    ambient    failed to find pod netns    ns=default name=details-v1-5f4d584748-ddk86                                                                                                                                  ??
2024-02-03T13:43:47.988730Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=details-v1-5f4d584748-ddk86                                                                                                ??
2024-02-03T13:43:47.988772Z    warn    ambient    failed to find pod netns    ns=default name=httpbin-worker1-7b47f9576c-m27vp                                                                                                                             ??
2024-02-03T13:43:47.988779Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=httpbin-worker1-7b47f9576c-m27vp                                                                                           ??
2024-02-03T13:43:47.988790Z    warn    ambient    failed to find pod netns    ns=default name=notsleep-5ccf57569-2h7xp                                                                                                                                     ??
2024-02-03T13:43:47.988798Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=notsleep-5ccf57569-2h7xp                                                                                                   ??
2024-02-03T13:43:47.988806Z    warn    ambient    failed to find pod netns    ns=default name=ratings-v1-686ccfb5d8-5xgbj                                                                                                                                  ??
2024-02-03T13:43:47.988810Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=ratings-v1-686ccfb5d8-5xgbj                                                                                                ??
2024-02-03T13:43:47.988815Z    warn    ambient    failed to find pod netns    ns=default name=sleep-9454cc476-lkvgc                                                                                                                                        ??
2024-02-03T13:43:47.988819Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=sleep-9454cc476-lkvgc                                                                                                      ??
2024-02-03T13:43:47.988855Z    warn    ambient    failed to find pod netns    ns=default name=backend-5cbb9cd947-z5lgn                                                                                                                                     ??
2024-02-03T13:43:47.988863Z    error    ambient    failed to remove pod to ztunnel: failed to find pod netns    ns=default name=backend-5cbb9cd947-z5lgn 

2024-02-03T13:44:37.726771Z    error    Command error output: xtables other problem: line 2 failed                                                                                                                                                         ??
2024-02-03T13:44:37.726786Z    error    iptables    failed to restore iptables rules: exit status 1                                                                                                                                                        ??
2024-02-03T13:44:37.726812Z    error    ambient    failed to update POD inpod: default/ratings-v1-686ccfb5d8-5xgbj exit status 1                                                                                                                           ??
2024-02-03T13:44:37.726817Z    error    ambient    failed to add pod to ztunnel: exit status 1  
```

RCA: while controller watch POD/NS event, based on the NS label event, we will add the pod into mesh, an annotation will be tagged to the pod by design, then another POD update event will be triggered, an unexpected Add/Remove action executed.

Fix: use latest pod status to determine if further action needed

### CODE REVIEW

**Comment by PlatformLC** [comment]
@bleggett PTAL, the UT failed should be expected we remove the 'junk' update. I'm not sure if it's intended to add such junk update, any background of it?

**Review by hzxuzhonghu — COMMENTED** [review]
please add tests

**Comment by PlatformLC** [comment]
> please add tests

Updated, PTAL

**Review by hzxuzhonghu — APPROVED** [review]
Thank you

File: cni/pkg/nodeagent/informers.go [hunk 1]
```diff
@@ -207,10 +207,16 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 		}
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
+		isOpOut := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionDisabled
+
```
Comment (bleggett) [inline cni/pkg/nodeagent/informers.go:211]: Why add this, when

`shouldBeEnabled := util.PodRedirectionEnabled(ns, newPod)`

already checks for `constants.AmbientRedirectionDisabled` in the line right after?

Comment (bleggett) [reply cni/pkg/nodeagent/informers.go:211]: Oh I see.

TY for finding and adding tests! We can actually simplify this a bit by weakening the `isAnnotated` check to just check for either label: https://github.com/istio/istio/pull/49209

**Pull request merged**